internal/provider/adc/adc.go

```diff
@@ -75,20 +175,80 @@ func (d *adcClient) Update(ctx context.Context, tctx *provider.TranslateContext,
 		return nil
 	}
 
+	// Update configs for non-HTTPRoute types
+	if _, ok := obj.(*gatewayv1.HTTPRoute); !ok {
+		if err := d.updateConfigs(rk, tctx); err != nil {
+			return err
+		}
+	}
+
 	return d.sync(Task{
 		Name:   obj.GetName(),
 		Labels: label.GenLabel(obj),
-		Resources: types.Resources{
+		Resources: adctypes.Resources{
 			GlobalRules:    result.GlobalRules,
 			PluginMetadata: result.PluginMetadata,
 			Services:       result.Services,
 			SSLs:           result.SSL,
 			Consumers:      result.Consumers,
 		},
 		ResourceTypes: resourceTypes,
+		configs:       d.getConfigs(rk),
```

Comment:
I think it would be more appropriate to handle it in one place.
For example, adding `deleteConfigs`